Tasks/AzurePowerShellV4/AzurePowerShell.ts

```diff
@@ -0,0 +1,137 @@
+                throw new Error(tl.loc('JS_InvalidFilePath', scriptPath));
+            }
+        }
+
+        // Generate the script contents.
+        console.log(tl.loc('GeneratingScript'));
+        let contents: string[] = [];
+        let azFilePath = path.join(path.resolve(__dirname), 'InitializeAz.ps1');
+        contents.push(`$ErrorActionPreference = '${_vsts_input_errorActionPreference}'`); 
+        if(targetAzurePs == "") {
+            contents.push(`${azFilePath} -serviceName ${serviceName} -endpoint '${endpoint}'`);
+        }
+        else {
+            contents.push(`${azFilePath} -targetAzurePs  ${targetAzurePs} -serviceName ${serviceName} -endpoint '${endpoint}'`);
+        }
+        
+        if (scriptType.toUpperCase() == 'FILEPATH') {
+            contents.push(`. '${scriptPath.replace("'", "''")}' ${scriptArguments}`.trim());
+            console.log(tl.loc('JS_FormattedCommand', contents[contents.length - 1]));
+        }
+        else {
+            contents.push(scriptInline);
+        }
+
+        // Write the script to disk.
+        tl.assertAgent('2.115.0');
+        let tempDirectory = tl.getVariable('agent.tempDirectory');
+        tl.checkPath(tempDirectory, `${tempDirectory} (agent.tempDirectory)`);
+        let filePath = path.join(tempDirectory, uuidV4() + '.ps1');
+
+        await fs.writeFile(
+            filePath,
+            '\ufeff' + contents.join(os.EOL), // Prepend the Unicode BOM character.
+            { encoding: 'utf8' });           // Since UTF8 encoding is specified, node will
+                                            // encode the BOM into its UTF8 binary sequence.
+
+        // Run the script.
+        //
+        // Note, prefer "pwsh" over "powershell". At some point we can remove support for "powershell".
+        //
+        // Note, use "-Command" instead of "-File" to match the Windows implementation. Refer to
+        // comment on Windows implementation for an explanation why "-Command" is preferred.
+        let powershell = tl.tool(tl.which('pwsh') || tl.which('powershell') || tl.which('pwsh', true))
+            .arg('-NoLogo')
+            .arg('-NoProfile')
+            .arg('-NonInteractive')
+            .arg('-ExecutionPolicy')
+            .arg('Unrestricted')
+            .arg('-Command')
+            .arg(`. '${filePath.replace("'", "''")}'`);
+
+        let options = <tr.IExecOptions>{
+            failOnStdErr: false,
+            errStream: process.stdout, // Direct all output to STDOUT, otherwise the output may appear out
+            outStream: process.stdout, // of order since Node buffers it's own STDOUT but not STDERR.
+            ignoreReturnCode: true
+        };
+
+        // Listen for stderr.
+        let stderrFailure = false;
+        if (_vsts_input_failOnStandardError) {
+            powershell.on('stderr', (data) => {
+                stderrFailure = true;
+            });
+        }
```

Comment:
We need to clear endpoint context

Reply:
Required only for SPN certificate, not for SPN credentials